lib/interval.js

```diff
@@ -70,14 +70,19 @@ Interval.prototype.assign = function (lo, hi) {
   if (typeof lo !== 'number' || typeof hi !== 'number') {
     throw TypeError('Interval#assign: arguments must be numbers')
   }
-  if (isNaN(lo) || isNaN(hi) || lo > hi) {
+  //if (isNaN(lo) || isNaN(hi) || lo > hi) {
+  if (isNaN(lo) || isNaN(hi)) {
     return this.setEmpty()
   }
+  if (lo > hi){ return this.set(hi, lo)} // turn [a,b] to [b,a] if a > b.
```

Comment:
From what I understand the interval n = Interval(next(1), 1) represents the semiclosed interval (1,1] or ]1,1] according to some other notation. This could be rewritten as [1+ε, 1], with ε>0 a very small number. So obviously 1+ε > 1 and since [0,1] \ [0,1] = \emptyset your definition of [1+ε, 1] as the empty set is very reasonable.

Perhaps we should distinct these extreme cases though where x.lo - x.hi = ε > 0, for some interval X = [x.lo, x.hi], with the mainstream cases where x.lo and x.hi differ by a bigger number than ε.

I think I have seen an example of interval calculation where one can get an interval with x.lo > x.hi that would make a good counterexample to your definition and show that perhaps that in most cases we shouldn't define such an interval as the empty set. I'll look in my notes and let you know.